src/status_im/chat/new_public_chat/db.cljs

```diff
@@ -0,0 +1,20 @@
+(ns status-im.chat.new-public-chat.db
+  (:require [cljs.spec.alpha :as spec]
+            [status-im.constants :refer [console-chat-id wallet-chat-id]]
+            [clojure.string :as string]
+            [status-im.utils.homoglyph :as utils]))
+
+(defn not-illegal-name? [username]
+  (let [username (some-> username (string/trim))]
+    (and (not (utils/matches username console-chat-id))
+         (not (utils/matches username wallet-chat-id)))))
+
+(spec/def ::not-empty-string (spec/and string? not-empty))
+(spec/def ::not-illegal-name not-illegal-name?)
+
+(spec/def ::name (spec/and ::not-empty-string
+                           ::not-illegal-name))
+
+(spec/def ::topic (spec/and ::not-empty-string
+                            ::not-illegal-name
+                            (partial re-matches #"[a-z0-9\-]+")))
```

Comment:
There is a lot of negation in predicates and spec definitions here. I would argue it's simpler and less confusing to define non-negated predicates and negate them in logical condition when needed:
```clj
(defn illegal-name? [username]
  (let [trimmed-username (some-> username string/trim)]
    (or (util/matches trimmed-username console-chat-id)
        (util/matches trimmed-username wallet-chat-id))))

(def not-empty-string? (comp string? not-empty))

(spec/def ::name (spec/and not-empty-string?
                           (complement illegal-name?)))
```

Also, it's matter of taste but I wouldn't define spec for things which are just sub-parts of spec like `::not-empty-string` and `::not-illegal-name` here. If they are serving as spec on their own (like some key in db is specced as `::not-empty-string`), then absolutely yes, but if they only serve as building block for another spec (like `::name` and `::topic`) then including them as simple functions is enough, less code and easier to follow.